### DISCUSSION TITLE
Issue with Beacon Node and Validator Deposits on Bedrock Devnet local

### DISCUSSION THREAD

**Discussion by doukhahmed** [ethereum-optimism/developers#599] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I’m running a local Bedrock devnet with three containers (L1, L2, and Beacon node). The containers have started correctly, but I am encountering issues related to validator deposits and the connection between the Beacon node and the execution layer (L1).

### Additional Information

Here are the details of the problem:
Issue with Validator Deposits: The logs from the Beacon node indicate that it is waiting for more validator deposits to reach the required threshold, but the deposits don't seem to be processed or detected. I currently see messages like:
**INFO Waiting for more deposits, valid_deposits: None, total_deposits: 0, min_genesis_active_validators: 64**
Connection between Beacon Node and L1: There appears to be a connection issue between the Beacon node and the execution layer (L1). The Beacon node logs show that it is trying to connect to the L1 endpoint, but it doesn’t seem to authenticate or establish a valid connection.
**WARN Running HTTP server on port 5052
INFO Waiting for eth2 genesis from eth1
deposit_contract: 0x1111111111111111111111111111111111111111
eth1_endpoints: Auth { endpoint: "http://l1:8551/", jwt_path: "/config/jwt-secret.txt" }**

The Beacon node should successfully connect to the execution layer (L1) and process validator deposits, allowing the Bedrock devnet to proceed to the genesis phase.

- Do I need to modify any configuration settings related to the validator deposits or the connection between the Beacon node and L1?

- Is there a known issue with the current Bedrock devnet setup that could be causing this problem?

- Are there any additional steps I should take to troubleshoot or resolve this issue?

**Comment by hash1go** [comment]
I'd like to understand how the hostname 'l1' in http://l1:8551/ is being resolved. Could you explain your current setup for name resolution between your Docker containers?
